manifests/deploy/export.pp

```diff
@@ -0,0 +1,96 @@
+# == Define: openvpn::deploy::export
+#
+# Prepare all Openvpn-Client-Configs to be exported
+#
+# === Parameters
+#
+# $server   which Openvpn::Server[$server] does the config belong to?
+# String
+#
+# $tls_auth should the ta* files be exported too?
+#
+# === Variables
+#
+# None
+#
+# === Examples
+#
+#  openvpn::deploy::export { 'test-client':
+#    server => 'test_server',
+#  }
+#
+# === Authors
+#
+# Tobias Knipping https://github.com/to-kn
+# Phil Bayfield https://bitbucket.org/Philio/
+#
+
+define openvpn::deploy::export (
+  String $server,
+  Boolean $tls_auth = false,
+) {
+
+  Openvpn::Server[$server]
+  -> Openvpn::Client[$name]
+  -> Openvpn::Deploy::Export[$name]
+
+  if $facts['openvpn::client_configs'] {
```

Comment:
on a first view, this looks like you lookup a puppet variable in the facts hash, I think we should change it.

Reply:
would you prefer something along the lines of "openvpn_client_configs"?

Reply:
openvpn_client_configs or just openvpn or openvpn {client_configs{}} sound good I guess